### DISCUSSION TITLE
Design Proposal: Alternative methods of discouraging "Perma-Suiting" 🤿

### DISCUSSION THREAD

**Discussion by T1MM5H** [FakeFishGames/Barotrauma#14075] [Ideas and suggestions]
_Tld;dr: Use Skill debuffs and adjust the Oxygen economy_

### Problem: Players are spending most of their time in Diving Suits, invalidating flooding as a threat
Being a game about submarining, flooding is a central mechanic in Barotrauma. However, it is also thus omnipresent, and players will spend most of their time in Diving Suits in order to stay prepared. The utility of a Diving Suit combined with their statistics has made this (until the "Blood in the Water" changes) the optimal way to play.

### Previous Solution: Disincentive Permasuiting by slowing down the player and toning down protection
The "Blood in the Water" update reduced the viability of wearing suits all the time by reducing the movement speed, visibility, and protection afforded to the player. (See: https://youtu.be/R2jZe843h8E?si=p4c82VUhGeQSY6YT for a visual breakdown, and shoutout to InitialDesu) 

In principle, these changes are effective, but they also frustrate the playerbase beyond just the simple knee-jerk reaction of "my toys got taken away". The main gripe seem to be:

_"Dead Air" and combat woes when using Suits_ <br> The changes, being what they are, also affected situations in which wearing a suit is appropriate, such as when scavenging a wreck or re-floating a damaged vessel. Barotrauma already has long periods of time in which the player completes relatively mundane tasks such as inventory management or ferrying two-handed items, and slowing all divers down only exasperates this problem. 

Additionally, on higher difficulties, both the Defense Debuffs and Movement speed reduction amplify eachother, having the player exposed to attack for longer (due to quick enemy spawn intervals) while also reducing their staying power. This leads to frustrations with Scavenging and EVA combat, some of the most fun aspects of Barotrauma!

_Just to clarify: I think the vision change was good, the movement nerf was a little too much, and the protection nerf FAR too much._

### Proposed Solution A: Skill penalties & Clumsyness
**In my opinion, this is the alternative i feel suits the game best at the moment. This does not mean it is the only way to tune the suit balance. The best solution will likely be a synthesis of multiple balance changes, because life is complicated like that.** 

This proposed solution would consist of an affliction i would call "Clumsiness", with a variable strength clamped from 0-100. Wearing a suit (and maybe some other equipment like belts) would apply this Affliction to varying degrees. <br>
The "Clumsy" affliction would decrease skills by an amount corresponding to it's current strength. So, at 50 Clumsy, the player would receive -50 to all skills. Here's how a stat distribution of "Clumsy" could look like:

- Disposable Diving Suit = +80 Clumsy
- Diving Suit = +60 Clumsy
- Combat Diving Suit = +50 Clumsy, +10 Weapons Bonus
- Slipsuit = +60 Clumsy, +20 Weapons Bonus
- PUCS = +40 Clumsy +20 EE +20 ME
- Backpack = +10 Clumsy
- Toolbelt = +5 Clumsy

An extra addition could include a chance to trip (ragdoll) while walking once over a certain threshold (50?). I would also propose adding a Common T1 Talent which reduces the Clumsy Debuff by a flat 20 points.

**Benefits:**

- Less boring. Due to emphasis on the skill debuff, wearing a suit is discouraged in normal situations without compromising "adventure" gameplay
- Promotes teamplay. In emergencies that require suits, players are encouraged to take on problems which their character is suited to solving, rather than the closest, and during EVA, bringing along trained personnel becomes more important
- Allows emphasizing suit roles. By tweaking the skill modifier on suits, they can be encouraged to fill roles that fit their description rather than just being upgrades to the default

**Drawbacks:**

- Early game emergencies will be much deadlier unless starting skill level is increased
- Will make certain weapons with high skill requirement difficult to use

### Proposed Solution B: Tuning the Oxygen economy
Since last year, the Oxygen Economy has remained largely untouched. However, i think by carefully tuning this aspect of the game one can discourage keeping a suit on without affecting fun factor too much.

In my opinion, the best way to tune oxygen is by decreasing the rate at which they fill in the various receptables. There's a couple ways this could be done:

1. Simply nerf oxygen refill rate. This makes suits less viable to wear all the time, as players will not have enough oxygen tank regeneration to sustain them indefinitely. <br> This blanket solution will not scale well at all. Subs with an abundance of oxygen refill stations will profit, while subs that have just a few will suffer. New subs will simply add more shelves to compensate.
2. Add a "Compressor Budget" to the Submarine statistics. This global variable would be increased by upgrades and additional generators and divided among currently active oxygen refill stations. <br> This solution would scale well and automatically balance itself, as smaller subs and crews will charge less tanks at once, leading to highter relative refill rates.
3. Add a "Compressor Budget" to the Oxygen Generators. This new statistic would determine refill rates in the same way as option 2, by giving each Oxygen Generator a Variable describing how much "compressed" oxygen it can deliver to canisters. <br> This change would be the most user-facing, user-friendly variant, but it would also immediately render most custom subs obsolete, because players would have to link oxygen tank shelves to oxygen generators. [Perhaps this could be avoided by moving the current oxygen tank shelves onto Legacy and implementing new shelves as a new object?]

**Benefits**
- Encourages Oxygen rationing. Players will be less likely to casually consume oxygen tanks in non-emergency or EVA situations, due to the possibility of being caught with a deficit
- Improves "Survival Horror" atmosphere. The sense of having precious, finite resources which you depend on to survive is a great driver for atmosphere and gameplay, and making oxygen less easily replaced would contribute.

**Drawbacks**
- Griefing. Griefing by deconstructing or dumping oxygen tanks will continue to be the most effective method of quiet griefing available to trolls

**Matters of Taste**
- Less hold-outs. Depending on preference, this might be a positive. Having less compressor capacity than the crew requires will create situations in which there is a strict time limit for creating a breathable atmosphere inside the submarine when it has been damaged. Captains may be required to execute the clown to save oxygen.

### Please give feedback/discuss!

**Comment by SomeRandomNoobKekeke** [comment]
I completed solo campaign without ever wearing the suits inside the sub, so it's a skill issue
![barodev small](https://github.com/FakeFishGames/Barotrauma/assets/122838333/e390e507-cd70-49b1-b42f-94aa876997e3)

**Reply by T1MM5H** [reply]
That's not really the point i was making? There's clearly a goal to steer the gameplay in that direction, it could be done in a way that's more organic, not neccessarily less difficult.

Edit: For context, there's a lot of people that will optimize their playstyle until the game becomes a trivial chore, this is a general problem in game design and it's usually a good idea to address it. You're just one of the types of players who like to challenge themselves by playing with self-imposed rules/RP, and that's valid, but i don't think the gameplay loop should rely on every player doing this.

**Comment by bookslayer10** [comment]
Interesting ideas. First, I don't think nerfing toolbelts and backpacks is good, they already have their own downsides. Toolbelts prevent you from carrying the best weapons in the game, and backpacks have a slowdown. Further skill nerfs are unnecessary.

I don't think the skill nerfs solve the problem you are trying to solve. You bring up how the current dive suit nerfs aren't fun because they effect the stuff you are supposed to use the dive suit for, but that's exactly what happens when you nerf skills while in them. It just makes dive suits much harder to do stuff in, which makes the game more frustrating, not even mentioning the further indirect buff to husk this would bring.

The oxygen changes are probably a better way to go, but I'm not sure how well a logistical challenge would go. If oxygen was a scarcer resource, then that could spark friction over people "wasting" oxygen or not using it efficiently enough. Making oxygen tanks refill slower would just make the ship logistics assistant get more frustrated juggling everyone's empty tanks.
